src/Chess/ChessPieces.hs

```diff
@@ -19,8 +19,39 @@ legalNextPosForPieceAtPos :: ChessPiece -> ChessBoard -> ChessPosition -> [Chess
 
 -- King can move exactly one square horizontally, vertically, or diagonally (ignore castling for now).
 -- King cannot move to a square if a piece of its own colour is blocking the way.
--- TODO: Implement + test this (moves for King at the given position) (1.5 hours) [Cynthia]
-legalNextPosForPieceAtPos (King colour) chessBoard position = []
+legalNextPosForPieceAtPos (King colour) chessBoard (col, row)  = 
```

Comment:
This looks theoretically right but in certain edge cases you might get issues with evaluating invalid values before you intend to. For example, even though Haskell does lazy evaluation, I found that in a statement like:

```haskell
if (row < 8) && (topEmpty || topHasEnemyPiece) then [(col, row+1)] else []
```
It might evaluate `topEmpty` before checking `(row < 8)`. It only happens sometimes - I'm not sure what triggers it. 

To make sure all the logic is sound, we should test all the cases. See my comment below about more exhaustive testing.

Reply:
I'm also wondering if it might be worth converting checks such as `row < 8` and `col /= 'A'` into separate functions - I used those in my code as well and I guess we will use it in other places too. Just if you feel like and find a good way to abstract it out, not crucial though.